src/commands/app/list-upgrade-candidates.ts

```diff
@@ -0,0 +1,68 @@
+import { Simplify } from "@mittwald/api-client-commons";
+import {
+  appInstallationArgs,
+  withAppInstallationId,
+} from "../../lib/app/flags.js";
+import { ListBaseCommand } from "../../ListBaseCommand.js";
+import { MittwaldAPIV2, MittwaldAPIV2Client } from "@mittwald/api-client";
+import { ListColumns } from "../../Formatter.js";
+import { getAppInstallationFromUuid } from "../../lib/app/uuid.js";
+
+type ResponseItem = Simplify<
+  MittwaldAPIV2.Paths.V2AppsAppIdVersions.Get.Responses.$200.Content.ApplicationJson[number]
+>;
+
+type Response = Awaited<
+  ReturnType<MittwaldAPIV2Client["app"]["listAppversions"]>
+>;
+
+export default class List extends ListBaseCommand<
+  typeof List,
+  ResponseItem,
+  Response
+> {
+  static description = "List update candidates for an appInstallation.";
+  static args = {
+    ...appInstallationArgs,
+  };
+  static flags = {
+    ...ListBaseCommand.baseFlags,
+  };
+
+  protected async getData(): Promise<Response> {
+    const appInstallationId = await withAppInstallationId(
+      this.apiClient,
+      List,
+      this.flags,
+      this.args,
+      this.config,
+    );
+
+    const currentAppInstallation = await getAppInstallationFromUuid(
+      this.apiClient,
+      appInstallationId,
+    );
+
+    return await this.apiClient.app.listUpdateCandidatesForAppversion({
+      appId: currentAppInstallation.appId,
+      baseAppVersionId: (
+        currentAppInstallation.appVersion as { current: string }
```

Comment:
Why is this typecast necessary?

Reply:
It isn't that is a inadequate quickfix I forgot to properly handle because I didn't know how at that time.
Follow up question to that one though: The problem, wich I didnt understand at that time, was just that `baseAppVersionId` expects a string, and `currentAppInstallation.appVersion.current` might be `undefined`. So the solutaion is a basic check if it is undefined.

Now to my question: Usually I would use `process.error()` and let the process handle the error.
In the `ListBaseCommand` we dont have a process though. Do I just `throw.error()`? (90d3c58abce4f3cc34817428761ae64beeac3e6a)

Reply:
> So the solution is a basic check if it is undefined.

Please note that the typecast does not _actually_ check if `.current` is undefined; you just tell the compiler that it's not (and it will fail at runtime, if it turns out to be undefined, anyway).

You could either assert that `.current` is indeed a string (`typeof currentAppInstallation.appVersion.current === "string"` should be sufficient for the compiler) and throw an error if not (the error will be handled properly by the framework itself);

Alternatively, you might also check against `.desired`, which IRRC is not-nullable -- which might just work better anyway, in case a user decides to invoke `list-upgrade-candidates` on an app installation for which an upgrade is currently in progress.